[glib] glib-2.0 .pc file contains non-existing include directory

**Describe the bug**
The file `installed/x64-linux/debug/lib/pkgconfig/glib-2.0.pc` contains a reference to a non-existing include directory.

**Environment**
- OS: Linux
- Compiler: Not Relevant

**To Reproduce**
Steps to reproduce the behavior:
1. ./vcpkg install gts:x64-linux
2. Inspect the file `installed/x64-linux/debug/lib/pkgconfig/glib-2.0.pc` . It contains a reference to ` -I"${includedir}/glib-2.0"` that expands to `installed/x64-linux/include/glib-2.0`  `-I"${libdir}/glib-2.0/include"`, that expands to  `installed/x64-linux/debug/lib/glib-2.0/include` . However, both directories do not exists, as the only directory that exists is ``installed/x64-linux/include/glib`.

**Expected behavior**
The `.pc`  file of glib should only contain references to existing directories.

**Failure logs** 
The PR https://github.com/microsoft/vcpkg/pull/11273 fails with the following error due to this bug:
~~~
CMake Error in graphics/src/CMakeLists.txt:
  Imported target "GTS::GTS" includes non-existent path

    "/mnt/vcpkg-ci/installed/x64-linux/debug/lib/pkgconfig/../../../include/glib-2.0"

  in its INTERFACE_INCLUDE_DIRECTORIES.  Possible reasons include:

  * The path was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and references files it does not
  provide.
~~~

**Additional details** 

Related PR: https://github.com/microsoft/vcpkg/pull/11884 .

--- a/ports/glib/CMakeLists.txt
+++ b/ports/glib/CMakeLists.txt
@@ -1,7 +1,6 @@
 cmake_minimum_required(VERSION 3.0)
 project(glib C)
 
-set(PACKAGE_VERSION 2.52.3)
 set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_C_STANDARD 11)
 
@@ -358,7 +357,7 @@ message(STATUS "  " ${CHARSET_LIBRARY})
 message(STATUS "  " ${FFI_LIBRARY})
 message(STATUS "  " ${Intl_LIBRARIES})
 
-set(VERSION ${PACKAGE_VERSION})
+set(VERSION ${GLIB_VERSION})
 set(prefix ${CMAKE_INSTALL_PREFIX})
 set(exec_prefix ${CMAKE_INSTALL_PREFIX})
 set(libdir ${CMAKE_INSTALL_PREFIX}/lib)

--- a/ports/glib/CMakeLists.txt
+++ b/ports/glib/CMakeLists.txt
@@ -1,6 +1,7 @@
 cmake_minimum_required(VERSION 3.0)
 project(glib C)
 
+set(PACKAGE_VERSION 2.52.3)
 set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_C_STANDARD 11)
 
@@ -357,6 +358,7 @@ message(STATUS "  " ${CHARSET_LIBRARY})
 message(STATUS "  " ${FFI_LIBRARY})
 message(STATUS "  " ${Intl_LIBRARIES})
 
+set(VERSION ${PACKAGE_VERSION})
 set(prefix ${CMAKE_INSTALL_PREFIX})
 set(exec_prefix ${CMAKE_INSTALL_PREFIX})
 set(libdir ${CMAKE_INSTALL_PREFIX}/lib)

--- a/ports/glib/CONTROL
+++ b/ports/glib/CONTROL
@@ -1,6 +1,6 @@
 Source: glib
 Version: 2.52.3
-Port-Version: 24
+Port-Version: 25
 Homepage: https://developer.gnome.org/glib/
 Description: Portable, general-purpose utility library.
 Build-Depends: zlib, pcre, libffi, gettext, libiconv

--- a/ports/glib/fix_pkgconfig.patch
+++ b/ports/glib/fix_pkgconfig.patch
@@ -1,0 +1,30 @@
+diff --git a/gio-unix-2.0.pc.in b/gio-unix-2.0.pc.in
+index fba58e2..2d015f6 100644
+--- a/gio-unix-2.0.pc.in
++++ b/gio-unix-2.0.pc.in
+@@ -8,4 +8,4 @@ Description: unix specific headers for glib I/O library
+ Version: @VERSION@
+ Requires: gobject-2.0,gio-2.0
+ Libs: -L${libdir} -lgio-2.0
+-Cflags: -I${includedir}/gio-unix-2.0/
++Cflags: -I${includedir}/gio
+diff --git a/gio-windows-2.0.pc.in b/gio-windows-2.0.pc.in
+index 77eecdf..25bb0ae 100644
+--- a/gio-windows-2.0.pc.in
++++ b/gio-windows-2.0.pc.in
+@@ -8,4 +8,4 @@ Description: Windows specific headers for glib I/O library
+ Version: @VERSION@
+ Requires: gobject-2.0,gmodule-no-export-2.0,gio-2.0
+ Libs: -L${libdir} -lgio-2.0
+-Cflags: -I${includedir}/gio-win32-2.0/
++Cflags: -I${includedir}/gio
+diff --git a/glib-2.0.pc.in b/glib-2.0.pc.in
+index 275fc01..1bfd74a 100644
+--- a/glib-2.0.pc.in
++++ b/glib-2.0.pc.in
+@@ -13,4 +13,4 @@ Version: @VERSION@
+ Requires.private: @PCRE_REQUIRES@
+ Libs: -L${libdir} -lglib-2.0 @INTLLIBS@
+ Libs.private: @G_THREAD_LIBS@ @G_LIBS_EXTRA@ @PCRE_LIBS@ @INTLLIBS@ @ICONV_LIBS@ @CARBON_LIBS@ @COCOA_LIBS@
+-Cflags: -I${includedir}/glib-2.0 -I${libdir}/glib-2.0/include @GLIB_EXTRA_CFLAGS@
++Cflags: -I${includedir}/glib @GLIB_EXTRA_CFLAGS@

--- a/ports/glib/portfile.cmake
+++ b/ports/glib/portfile.cmake
@@ -20,6 +20,7 @@ vcpkg_extract_source_archive_ex(
         use-libiconv-on-windows.patch
         arm64-defines.patch
         fix-arm-builds.patch
+        fix_pkgconfig.patch
 )
 
 configure_file(${CMAKE_CURRENT_LIST_DIR}/CMakeLists.txt  ${SOURCE_PATH}/CMakeLists.txt COPYONLY)

